utils/python/CIME/XML/entry_id.py

```diff
@@ -15,32 +17,60 @@ def __init__(self, infile=None):
         GenericXML.__init__(self, infile)
         self.groups={}
 
-    def set_default_value(self, node, attributes=None):
+    def get_default_value(self, node, attributes={}):
         """
         Set the value of an entry to the default value for that entry
         """
-        # Hangle this case:
+        value = self._get_value_match(node, attributes)
+        vid = node.get("id")
+        if value is None:
+            # Fall back to default value
+            val_node = self.get_optional_node("default_value", root=node)
+            if val_node is not None:
+                value = val_node.text
+        else:
+            logger.debug("node is %s value is %s"%(node.get("id"),value))
+        if value is None:
+            value = ""
+        return value
+
+    def get_value_match(self, vid, attributes={}):
+        # Handle this case:
         # <entry id ...>
         #  <values>
         #   <value A="a1">X</value>
         #   <value A="a2">Y</value>
-        #   <value A="a3">Z</value>
+        #   <value A="a3" B="b1">Z</value>
         #  </values>
         # </entry>
-        valnodes = self.get_nodes("value", root=node)
-        for valnode in valnodes:
-            for att in valnode.keys():
-                if att.key in attributes:
-                    if re.search(attributes[att.key], att.text):
-                        node.set("value", valnode.text)
-                        logger.debug("id %s value %s" % (node.get("id"), valnode.text))
-                        return valnode.text
-
-        # Fall back to default value
-        value = self.get_optional_node("default_value", root=node)
-        if value is not None:
-            node.set("value", value.text)
-            return value.text
+
+        node = self.get_optional_node(vid)
+        value = None
+        if node is not None:
+            value = self._get_value_match(self,node,attributes)
+        return value
+
+    def _get_value_match(self, node, attributes={}):
```

Comment:
get value match finds the xml entry with the highest count of matching attributes.